File: python_implementation/logging/ConsoleLogger.cs
using System;
using System.Collections.Generic;
using System.Text;
using Codecool.Logging;

namespace Codecool.MainProduct.Logging
{
    /// <summary>
    /// An implementation of the <see cref="ILogger"/> interface.
    /// Logs messages for the standard error stream of the process.
    /// </summary>
    public class ConsoleLogger : AbstractLogger
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="ConsoleLogger"/> class.
        /// </summary>
        /// <param name="levels">Log levels.</param>
        public ConsoleLogger(params LogLevel[] levels) : base(levels) { }

        /// <inheritdoc />
        protected override void SafeLog(LogLevel level, string message)
        {
            string formattedMessage = FormatMessage(level, message);
            Console.WriteLine(formattedMessage);
        }

        private string FormatMessage(LogLevel level, string message)
        {
            var time = DateTime.Now.ToUniversalTime();
            return $"{time} - {level.ToString()}: {message} ";
        }
    }
}

File: somnium/logger/console_logger.py
#!/usr/bin/env python

from typing import Callable
from log_level import LogLevel
from abstract_logger import AbstractLogger
import datetime


class ConsoleLogger(AbstractLogger):
    """
    An implementation of the ILogger interface.
    Logs messages for the standard error stream of the process.
    """

    def safe_log(self, level: LogLevel, message: str) -> None:
        """
        Log in a safe manner.
        """
        formatted_message = self.__format_message(level, message)
        print(formatted_message)

    def __format_message(self, level: LogLevel, message: str) -> str:
        time = datetime.datetime.now()
        return f"{time} - {level.name}: {message}"


